lib/opts.nit

```diff
@@ -147,12 +147,25 @@ abstract class OptionParameter
 	protected fun convert(str: String): VALUE is abstract
 
 	# Is the parameter mandatory?
-	var parameter_mandatory: Bool = true is writable
+	var parameter_mandatory = true is writable
 
-	redef fun read_param(it)
+	redef fun read_param(opts, it)
 	do
 		super
-		if it.is_ok and (it.item.is_empty or it.item.chars.first != '-') then
+
+		var ok = it.is_ok
+		if ok and not it.item.is_empty and it.item.chars.first == '-' then
+			# The next item may look like a known command
+			for opt in opts.options do
+				if opt.names.has(it.item) then
+					# The next item is a known command
+					ok = false
+					break
+				end
+			end
+		end
```

Comment:
Reroll to avoid known arguments on `mandatory_parameter == false` only. As such, it is not the default behavior but it is still available to those who really want it.